docs/rules/dot-notation.md

```diff
@@ -34,3 +47,31 @@ Set the `allowKeywords` option to `false` (default is `true`) to follow ECMAScri
 var foo = { "class": "CS 101" }
 var x = foo["class"]; // Property name is a reserved word, square-bracket notation required
 ```
+
+#### `allowPattern`
+
+Set the `allowPattern` option to a regular expression string to allow bracket notation for property names that match a pattern (by default, no pattern is tested).
+
+For example, when preparing data to be sent to an external API, it is often required to use property names that include underscores.  If the `camelcase` rule is in effect, these [snake case](http://en.wikipedia.org/wiki/Snake_case) properties would not be allowed.  By providing an `allowPattern` to the `dot-notation` rule, these snake case properties can be accessed with bracket notation.
+
+Example configuration:
+```js
```

Comment:
Addressed in https://github.com/tschaub/eslint/commit/7f0df57b99efb771145a20e63772b362ac42e378.